src/dmd/traits.d

```diff
@@ -1432,7 +1434,8 @@ Expression semanticTraits(TraitsExp e, Scope* sc)
         auto sds = s.isScopeDsymbol();
         if (!sds || sds.isTemplateDeclaration())
         {
-            e.error("%s `%s` has no members", s.kind(), s.toChars());
+            e.error("In expression `%s` %s `%s` has no members", e.toChars(), s.kind(), s.toChars());
```

Comment:
`template XXX has no members` could be confusing for some users. Should this emphazise that he passed a template declaration instead of a template instance?

Not a change request, just something worth evaluating)

Reply:
Yes, I don't like it too. I think that `kind()` method should be enhanced in other PR.

Reply:
Probably the easier way would be to add something like this (in case of template of course) - `did you pass a template declaration instead of an instatiated template?`
That can be implemented in this PR